api/base/serializers.py

```diff
@@ -55,24 +56,34 @@ def _url_val(val, obj, serializer, **kwargs):
 
 
 class IDField(ser.CharField):
+    """
+    ID field that validates that 'id' in the request body is the same as the instance 'id' for single requests
+    """
     def __init__(self, **kwargs):
         kwargs['label'] = 'ID'
         super(IDField, self).__init__(**kwargs)
 
     def to_internal_value(self, data):
         update_methods = ['PUT', 'PATCH']
-        if self.context['request'].method in update_methods:
+        request = self.context['request']
+        if request.method in update_methods and not isinstance(request.data, list):
```

Comment:
json-api says that you have to send `id` and `type` in the body of the request if you're updating a node.   This check ensures that the `id` sent in the request body matches the root's instance `id` (Node id).  So if I sent a PUT request to /v2/nodes/12345 with an `id`='ABCDE' in the body, a `409 Conflict` would be raised. 

However, when you're updating a list, the root's instance is a MongoQuerySet of all of nodes I want to update, instead of one Node.  I don't know if there's much to be gained from validating that the `id` sent in the request matches one of the id's in the MongoQuerySet.  The MongoQuerySet was formed from the id's in the request to begin with.